agent/hcp/deps.go

```diff
@@ -53,50 +56,48 @@ func NewDeps(cfg config.CloudConfig, logger hclog.Logger) (Deps, error) {
 	}, nil
 }
 
-// sink provides initializes an OTELSink which forwards Consul metrics to HCP.
+// sink initializes an OTELSink which forwards Consul metrics to HCP.
 // The sink is only initialized if the server is registered with the management plane (CCM).
-// This step should not block server initialization, so errors are logged, but not returned.
+// This step should not block server initialization, errors are returned, only to be logged.
 func sink(
 	ctx context.Context,
 	hcpClient hcpclient.Client,
-	metricsClient hcpclient.MetricsClient,
-	cfg config.CloudConfig,
-) metrics.MetricSink {
+	metricsClient telemetry.MetricsClient,
+) (metrics.MetricSink, error) {
 	logger := hclog.FromContext(ctx).Named("sink")
 	reqCtx, cancel := context.WithTimeout(ctx, 5*time.Second)
 	defer cancel()
 
 	telemetryCfg, err := hcpClient.FetchTelemetryConfig(reqCtx)
 	if err != nil {
-		logger.Error("failed to fetch telemetry config", "error", err)
-		return nil
+		return nil, fmt.Errorf("failed to fetch telemetry config: %w", err)
```

Comment:
I think this creates a situation where if the first call on start up fails we don't start the periodic refresh. The side-effect being that if HCP is down on agent start up, it won't continue querying its config and pick up a valid config when HCP service is restored. My thought is that instead we want a default, fall-back configuration here in case that first fetch fails.

Regarding the default, a couple fields are tricky but doable:
- filters: single wildcard
- labels: need to add cluster, project, org as we do right now (note: this creates a bit of a long-term contract HCP-side)
- endpoint: need to make this based on the environment

Reply:
Talked with @Achooo and she proposed a better idea where, rather than guessing defaults, we just return empty values from the config providers. I took a stab at implementing that, and I think it's a pretty small change given existing changes in the PR: https://github.com/hashicorp/consul/pull/18308/files

Reply:
If we do this in a follow-up, or decide not to start the sink if the first call fails, I'm fine w/ either. Gonna 🚢 the PR because I'll be OOO tomorrow, don't think this is blocking, and think what's here is a big improvement regardless